save-preprocessor/src/main/kotlin/org/cqfn/save/preprocessor/controllers/DownloadProject.kt

```diff
@@ -20,35 +20,36 @@ class DownloadProject {
     private val log = LoggerFactory.getLogger(DownloadProject::class.java)
 
     /**
-     * @param gitRepoDto - Dto of repo information to clone
+     * @param gitRepository - Dto of repo information to clone
+     * @return response entity with text
      */
+    @Suppress("TooGenericExceptionCaught")
     @PostMapping(value = ["/upload"])
-    fun upload(@RequestBody gitRepoDto: GitRepoDto): ResponseEntity<String> {
-        val user = if (gitRepoDto.username != null && gitRepoDto.password != null) {
-            UsernamePasswordCredentialsProvider(gitRepoDto.username, gitRepoDto.password)
+    fun upload(@RequestBody gitRepository: GitRepository): ResponseEntity<String> {
+        val tmpDir = File(VOLUMES)
+        tmpDir.deleteRecursively()
+        val user = if (gitRepository.username != null && gitRepository.password != null) {
+            UsernamePasswordCredentialsProvider(gitRepository.username, gitRepository.password)
         } else {
             CredentialsProvider.getDefault()
         }
         try {
             Git.cloneRepository()
-                .setURI(gitRepoDto.url)
+                .setURI(gitRepository.url)
                 .setCredentialsProvider(user)
-                .setDirectory(File(volumes))
+                .setDirectory(tmpDir)
                 .call().use {
-                    log.info("repository cloned")
+                    log.info("Repository cloned: ${gitRepository.url}")
                     // TODO post request to orchestrator
                     return ResponseEntity.ok("Cloned")
                 }
-        } catch (ex: TransportException) {
-            log.warn("No credentials with private repo")
-            return ResponseEntity.ok("No credentials")
         } catch (ex: Exception) {
-            log.warn("Error to clone repo")
-            return ResponseEntity.ok("Error to clone")
+            log.warn(ex.stackTraceToString())
+            return ResponseEntity.ok(ex.stackTraceToString())
         }
     }
 
     companion object {
-        private const val volumes = "~/var/lib/docker/volumes/repository"
+        private const val VOLUMES = "~/var/lib/docker/volumes/repository"
```

Comment:
This can be placed in properties file, so that you can override it in tests and check if the service stores data using some temporary directory.